### PR TITLE
137599-generate-project-template-transfers

### PR DESCRIPTION
**Pull request by danielryannimble**
new template added with placeholders
new code for document creation
unit tests updated

### CODE REVIEW

File: Dfe.PrepareTransfers.DocumentGeneration.Tests/DocumentBuilderTests.cs [hunk 1]
```diff
@@ -864,6 +879,25 @@ public void
                 Assert.Equal("", documentText[2]);
             }
 
+              [Fact]
+            public void
+                Givengene()
```
Comment (dneed-nimble) [inline Dfe.PrepareTransfers.DocumentGeneration.Tests/DocumentBuilderTests.cs:884]: What's this?

File: Dfe.PrepareTransfers.DocumentGeneration.Tests/Helpers/HtmlToDocumentTests.cs [hunk 2]
```diff
@@ -399,6 +399,11 @@ public void AddHeading(Action<IHeadingBuilder> action)
             throw new NotImplementedException();
         }
 
+        public void AddTextHeading(string headingText, HeadingLevel headingLevel)
+        {
+            throw new NotImplementedException();
```
Comment (dneed-nimble) [inline Dfe.PrepareTransfers.DocumentGeneration.Tests/Helpers/HtmlToDocumentTests.cs:404]: Can we either implement this or remove it if it isn't needed

File: Dfe.PrepareTransfers.Web/Services/DocumentGenerators/RisksGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                build.AddTextHeading("Risks", HeadingLevel.One);
+
+                if (projectTemplateModel.AnyIdentifiedRisks.Equals(true) )
+                {
+                      foreach (var item in projectTemplateModel.ListOfOtherFactors)
+                    {
+
+                        build.AddTable(new List<TextElement[]>
+                        {
+                         new[] { new TextElement { Value = EnumHelpers<TransferBenefits.OtherFactor>.GetDisplayValue(item.Key), Bold = true }, new TextElement { Value = item.Value } },
```
Comment (dneed-nimble) [inline Dfe.PrepareTransfers.Web/Services/DocumentGenerators/RisksGenerator.cs:25]: I think it's better to not have a new table for each risk, rather a new row for each risk. In summary if you move the foreach down to the text element level.

File: Dfe.PrepareTransfers.Web/Services/DocumentGenerators/BenefitsGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                }
+
+                else
+                {
+                    foreach (var item in projectTemplateModel.ListOfTransferBenefits)
+                    {
+
+                       var displayedOption = item == TransferBenefits.IntendedBenefit.Other ? $"Other: {projectTemplateModel.OtherIntendedBenefit}" : EnumHelpers<TransferBenefits.IntendedBenefit>.GetDisplayValue(item);
+
+                        build.AddTable(new List<TextElement[]>
```
Comment (dneed-nimble) [inline Dfe.PrepareTransfers.Web/Services/DocumentGenerators/BenefitsGenerator.cs:38]: Same as I said for risks below, move foreach down, saves space in doc

File: Dfe.PrepareTransfers.Web/Services/DocumentGenerators/RisksGenerator.cs (hunk 3 above)
Comment (dneed-nimble) [reply Dfe.PrepareTransfers.Web/Services/DocumentGenerators/RisksGenerator.cs:25]: Ah I've just seen Dan G's screenshots and it renders as rows regardless, so I'm happy with that.

File: Dfe.PrepareTransfers.Web/Services/DocumentGenerators/BenefitsGenerator.cs (hunk 4 above)
Comment (dneed-nimble) [reply Dfe.PrepareTransfers.Web/Services/DocumentGenerators/BenefitsGenerator.cs:38]: Ah I've just seen Dan G's screenshots and it renders as rows regardless, so I'm happy with that.

File: Dfe.PrepareTransfers.DocumentGeneration.Tests/Helpers/HtmlToDocumentTests.cs (hunk 2 above)
Comment (danielryannimble) [reply Dfe.PrepareTransfers.DocumentGeneration.Tests/Helpers/HtmlToDocumentTests.cs:404]: i think its required due to the interface that it inherits .. sonar cube was insiting it needs to be there

Comment (dneed-nimble) [reply Dfe.PrepareTransfers.DocumentGeneration.Tests/Helpers/HtmlToDocumentTests.cs:404]: Remove it from the interface?

File: Dfe.PrepareTransfers.DocumentGeneration.Tests/DocumentBuilderTests.cs (hunk 1 above)
Comment (danielryannimble) [reply Dfe.PrepareTransfers.DocumentGeneration.Tests/DocumentBuilderTests.cs:884]: deleted

**Review by dneed-nimble — APPROVED** [review]

**Pull request merged**